src/extractor.rs

```diff
@@ -19,41 +20,26 @@ pub struct Extraction {
 impl Arbitrary for Extraction {
     fn arbitrary(g: &mut Gen) -> Extraction {
         Extraction {
-            refs: Vec::<(String, Option<String>)>::arbitrary(g),
+            refs: Vec::<DbtRef>::arbitrary(g),
             sources: Vec::<(String, String)>::arbitrary(g),
             configs: Vec::<(String, ConfigVal)>::arbitrary(g),
         }
     }
 
     fn shrink(&self) -> Box<dyn Iterator<Item = Extraction>> {
-        let x: Vec<Vec<Extraction>> = vec![
-            self.configs
-                .shrink()
-                .map(|shrunk_config| Extraction {
-                    refs: self.refs.clone(),
-                    sources: self.sources.clone(),
-                    configs: shrunk_config,
-                })
-                .collect(),
-            self.refs
+        Box::new(
+            (
+                self.refs.clone(),
+                self.sources.clone(),
+                self.configs.clone(),
+            )
                 .shrink()
```

Comment:
This is very clever. I tested it locally and it is absolutely equivalent. I'm inclined to leave a comment here but that's up to you if you think that's necessary.